.github/mergify.yml

```diff
@@ -12,6 +12,58 @@ pull_request_rules:
       comment:
         message: This pull request has merge conflicts. Could you please resolve them
           @{{author}}? 🙏
+      label:
+        add:
+          - waiting-on-author
+        remove:
+          - ready-for-review
+
+  - name: Ask to resolve CI failures
+    conditions:
+      - or:
+          - check-failure=test-suite-success
+          - check-skipped=test-suite-success
+          - check-failure=local-testnet-success
+          - check-skipped=local-testnet-success
+    actions:
+      comment:
+        message: Some required checks have failed. Could you please take a look @{{author}}? 🙏
+      label:
+        add:
+          - waiting-on-author
+        remove:
+          - ready-for-review
+
+  - name: Update labels when PR is unblocked
+    conditions:
+      - label=waiting-on-author
+      - -conflict
+      - check-success=test-suite-success
+      - check-success=local-testnet-success
+      - "#changes-requested-reviews-by = 0"
```

Comment:
I think this won't fire very often. If someone requests changes, their request for changes hangs around until they dismiss it or re-review.

We can give it a go, but I think what we really need is a way of checking whether all review comments have been addressed. Maybe `#review-threads-unresolved = 0`?

Reply:
(from https://docs.mergify.com/configuration/conditions/)

Reply:
Good catch.

I had a look at a few open PRs and looks like neither of these would be sufficient
- `#changes-requested-reviews-by = 0` has the issue you mentioned, e.g. #7321 has a change requested so it will never fire until a maintainer changes the review status
- `#review-threads-unresolved = 0` may cause this rule to fire unintentionally, e.g. #7436 just had a change requested, but CI is passing and there's no conflict, therefore mergify will immediately remove the `waiting-on-author` label after it's added.

Reply:
For PRs by external contributors, this rule will never fire automatically because the CI workflow require approval - but it would be useful to automatically update the labels once the a maintainer approve the job and the CI job passes.

For PRs by maintainers, I think it would be easy enough for the author just update the labels.